src/rules/block-no-unnecessary-nesting/__tests__/index.js

```diff
@@ -0,0 +1,183 @@
+"use strict";
+
+const { messages, ruleName } = require("..");
+
+testRule({
+  ruleName,
+  config: true,
+  customSyntax: "postcss-scss",
+  fix: true,
+
+  accept: [
+    {
+      code: `
+        .foo .bar {}
+        .foo-bar {}
+        .foo {
+          color: red;
+        }
+      `,
+      description: "no nesting"
+    },
+    {
+      code: `
+        .foo {
+          .bar {}
+          &-baz {}
+        }
+      `,
+      description: "multiple child blocks"
+    },
+    {
+      code: `
+        .foo,
+        .bar {
+            &-baz {}
+        }
+      `,
+      description: "multiple outer selectors"
+    },
+    {
+      code: `
+        .foo {
+          .bar, .baz {}
+        }
+      `,
+      description: "multiple inner selectors"
+    },
+    {
+      code: `
+        .foo {
+            color: red;
+
+            &-baz {}
+        }
+      `,
+      description: "property and single child block"
+    },
+    {
+      code: `
+        .foo {
+            &-baz {}
+
+            @media (min-width: 10px) {}
+        }
+      `,
+      description: "at-rule and single child block"
+    }
+  ],
+
+  reject: [
+    {
+      code: `
+        .foo {
+          .bar {}
+        }
+      `,
+      fixed: `
+        .foo .bar {
+        }
+      `,
+      description: "single child block without &-prefix",
+      warnings: [
+        {
+          line: 3,
+          column: 11,
+          endLine: 3,
+          endColumn: 18,
+          message: messages.rejected
+        }
+      ]
+    },
+    {
+      code: `
+        .foo {
+          &-bar {
+            color: red;
+          }
+        }
+      `,
+      fixed: `
+        .foo-bar {
+            color: red
+        }
+      `,
+      description: "single child block with &-prefix and property",
+      warnings: [
+        {
+          line: 3,
+          column: 11,
+          endLine: 5,
+          endColumn: 12,
+          message: messages.rejected
+        }
+      ]
+    },
+    {
+      code: `
+        .foo:not(.bar, .baz) {
+          .bar {}
+        }
+      `,
+      fixed: `
+        .foo:not(.bar, .baz) .bar {
+        }
+      `,
+      description: "outer selector contains comma but is not a list",
+      warnings: [
+        {
+          line: 3,
+          column: 11,
+          endLine: 3,
+          endColumn: 18,
+          message: messages.rejected
+        }
+      ]
+    },
+    {
+      code: `
+        .foo {
+          .bar {}
+        }
+        .foo {
+          &-bar {
+            &-baz {
+              color: red;
+            }
+          }
+        }
+      `,
+      fixed: `
+        .foo .bar {
+        }
+        .foo-bar-baz {
+              color: red
```

Comment:
Small thing, but it would nice if the autofix would indent the properties correctly :)

Reply:
Well, actually, the core stylelint package has removed all styling related rules, so maybe the indentation is something that we want to leave to tools like prettier instead (check https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-15.md#deprecated-stylistic-rules)